presentation-compiler/src/main/dotty/tools/pc/printer/ShortenedTypePrinter.scala

```diff
@@ -0,0 +1,546 @@
+        s"${keyString(o)} $name: $ownerTypeString"
+      case m if m.is(Flags.Method) =>
+        defaultMethodSignature(m, info)
+      case _ =>
+        val implicitKeyword =
+          if sym.is(Flags.Implicit) then List("implicit") else Nil
+        val finalKeyword = if sym.is(Flags.Final) then List("final") else Nil
+        val keyOrEmpty = keyString(sym)
+        val keyword =
+          if keyOrEmpty.iterator.nonEmpty then List(keyOrEmpty) else Nil
+        (implicitKeyword ::: finalKeyword ::: keyword ::: (s"$name:" :: shortTypeString :: Nil))
+          .mkString(" ")
+    end match
+  end hoverSymbol
+
+  def completionSymbol(sym: Symbol): String =
+    val info = sym.info.widenTermRefExpr
+    val typeSymbol = info.typeSymbol
+
+    if sym.is(Flags.Package) || sym.isClass then " " + fullNameString(sym.owner)
+    else if sym.is(Flags.Module) || typeSymbol.is(Flags.Module) then
+      if typeSymbol != NoSymbol then " " + fullNameString(typeSymbol.owner)
+      else " " + fullNameString(sym.owner)
+    else if sym.is(Flags.Method) then
+      defaultMethodSignature(sym, info, onlyMethodParams = true)
+    else tpe(info)
+
+  /**
+   * Compute method signature for the given (method) symbol.
+   *
+   * @return shortend name for types or the type for terms
+   *         e.g. "[A: Ordering](a: A, b: B): collection.mutable.Map[A, B]"
+   *              ": collection.mutable.Map[A, B]" for no-arg method
+   */
+  def defaultMethodSignature(
+      gsym: Symbol,
+      gtpe: Type,
+      onlyMethodParams: Boolean = false,
+      additionalMods: List[String] = Nil
+  ): String =
+    val namess = gtpe.paramNamess
+    val infoss = gtpe.paramInfoss
+    val nameToInfo: Map[Name, Type] = (for
+      pairs <- namess.zip(infoss).map { (names, infos) => names.zip(infos) }
+      pair <- pairs
+    yield pair).toMap
+
+    val (methodParams, extParams) = splitExtensionParamss(gsym)
+    val paramss = methodParams ++ extParams
+    lazy val implicitParams: List[Symbol] =
+      paramss.flatMap(params => params.filter(p => p.is(Flags.Implicit)))
+
+    lazy val implicitEvidenceParams: Set[Symbol] =
+      implicitParams
+        .filter(p => p.name.toString.startsWith(EvidenceParamName.separator))
+        .toSet
+
+    lazy val implicitEvidencesByTypeParam: Map[Symbol, List[String]] =
+      constructImplicitEvidencesByTypeParam(
+        implicitEvidenceParams.toList
+      )
+
+    lazy val paramsDocs =
+      symbolSearch.symbolDocumentation(gsym) match
+        case Some(info) =>
+          (info.typeParameters.asScala ++ info.parameters.asScala).toSeq
+        case _ =>
+          Seq.empty
+
+    def label(paramss: List[List[Symbol]]) = {
+      var index = 0
+      paramss.flatMap { params =>
+        val labels = params.flatMap { param =>
+          // Don't show implicit evidence params
+          // e.g.
+          // from [A: Ordering](a: A, b: A)(implicit evidence$1: Ordering[A])
+          // to   [A: Ordering](a: A, b: A): A
```

Comment:
There's similar logic in https://github.com/lampepfl/dotty/blob/5c66d7b49421e0b7d399bcb506361b13c73db2ce/compiler/src/dotty/tools/dotc/util/Signatures.scala#L379, this can be done later but we should try to see if we can reuse Signatures here too.